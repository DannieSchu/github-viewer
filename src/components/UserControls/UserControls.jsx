import React from 'react';
import PropTypes from 'prop-types';

const UserControls = ({ username, onChange, onSubmit }) => {
  return (
    <>
      <form onSubmit={onSubmit}>
        <input type="text" placeholder="Username" value={username} onChange={onChange} />
        <button>Get Profile</button>
      </form>
    </>
  );
};

UserControls.propTypes = {
  username: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default UserControls;
