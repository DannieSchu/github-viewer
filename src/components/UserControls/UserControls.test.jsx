import React from 'react';
import { shallow } from 'enzyme';
import UserControls from './UserControls.jsx';

describe('UserControls', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <UserControls 
        username='DannieSchu' 
        onNameChange={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
