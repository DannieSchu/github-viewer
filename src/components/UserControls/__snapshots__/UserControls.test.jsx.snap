// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserControls matches a snapshot 1`] = `
<Fragment>
  <form>
    <input
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value="DannieSchu"
    />
    <button>
      Get Profile
    </button>
  </form>
</Fragment>
`;
